features/update/deduplication/feature.go

```diff
@@ -0,0 +1,96 @@
+package deduplication
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/features/features/update/updateutil"
+	"go.temporal.io/features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	reusedUpdateID = "reused_update_id"
+	incrementCount = "incrementCount"
+	shutdownSignal = "shutdown_signal"
+	expectedCount  = 1
+)
+
+var Feature = harness.Feature{
+	Workflows: Count,
+	Execute: func(ctx context.Context, runner *harness.Runner) (client.WorkflowRun, error) {
+		if reason := updateutil.CheckServerSupportsUpdate(ctx, runner.Client); reason != "" {
+			return nil, runner.Skip(reason)
+		}
+		run, err := runner.ExecuteDefault(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		handle1, err := runner.Client.UpdateWorkflowWithOptions(
```

Comment:
Since we are using the handle API is the intention for the updates to run in parallel? Because right now I don't think they will

Reply:
By adding the Sleep() in the middle of the update and using handlers I was trying to show that deduplication works even if the previous update has not completed yet. Unfortunately, you are right, there is no overlap, even if I use different IDs and different handler functions the server serializes all the updates for a workflow, i.e., one starts after the previous one completes. 

I couldn't find in the docs that this is expected behavior, but it has big implications for the programmers abstraction.  Reentrant update calls will deadlock, i.e., calling an update from an activity triggered by an update, but there are no races, which is really nice... 

In any case the docs should be explicit about this...

Reply:
I think the reason your seeing update requests is because in `UpdateWorkflowWithOptions` you didn't set a wait policy so I think it will default to `UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED` so your waiting for the whole request to complete when you send the update, not when your waiting on the handle result

Reply:
Thanks, once I set it to ACCEPTED they start to overlap (with different ids). 
What is very confusing is that in go the default is COMPLETED and  in java the default is ACCEPTED. We need to agree on one default...